Content Translations + a few fixes for old CMS packet (#18834)
directus/directus: docs/guides/headless-cms/content-translations.md

---
description: This guide shows how to setup content translations for internationalization (i18n).
tags: []
skill_level:
directus_version: 10.2.0
author_override:
author: Bryant Gillespie
---

# Creating Content Translations

> {{ $frontmatter.description }}

:::tip Author: {{$frontmatter.author}}

**Directus Version:** {{$frontmatter.directus_version}}

:::

Here's what the end result will look like for users when using the Data Studio to manage content.

> ![A Directus item detail page is shown using the Translations interface. Editable fields for the English and French translations for an article are displayed side-by-side. Fields include Title, Slug, Summary, and Content. ](https://marketing.directus.app/assets/64eee60e-605b-4d2c-809f-83557ed0d5ef.png?key=doc)

Reviewer: Great alt text

## Before You Start

You will need:

- A Directus project. The easiest way to get started with Directus is with our
  [managed Directus Cloud service](https://directus.cloud). You can also self-host Directus.
- Familiarity with [creating collections](/app/data-model/collections#create-a-collection) and
  [creating fields](/app/data-model/fields#create-a-field-standard) inside Directus.

> 

Reviewer: Maybe add a summary of the approach at this point.

> ## Create Languages Collection

Reviewer: ```suggestion
## Create A Languages Collection
```

> First, you'll need to create a new collection for all the different languages you'll support.

Reviewer: ```suggestion
Create a new collection for all the different languages your application will support.
```

   b. Add another field for the `name` of the language.

   c. Add one last field for the `direction` of the text if you're supporting languages that read right to left.

Now you're ready to move on.

## Create Content Collection

Before you can create translations, you need a collection of content to translate.

2. Create a new collection for `articles`.

   Here's an overview of the initial schema for this collection.

   ```md
   articles

   - id (Primary Key Field, Type: uuid )
   - status (Type: String, Interface: Dropdown, Options: ['published','draft','archived'])
   - image (Type: Image)
   - author (Type: M2O Relation, Related Collection: directus_users)
   - date_published (Type: Timestamp, Interface: Datetime)
   ```

   ![Data Model settings screen for the Articles collection is displayed. The following fields are shown: id, status, date_published, author, image.](https://marketing.directus.app/assets/fdeb07d9-d11a-42e9-a39a-14f0a196ecc6.png?key=doc)

>    Create the fields above inside the collection.

Reviewer: Reads weird.

   They will be shared across all the different translations of each article.

>    Do not add the fields to be translated like `title` or `content` or `slug` yet, because those will be stored in a
>    seperate collection you will create next.

Reviewer: ```suggestion
   Do not add the fields to be translated like `title` or `content` or `slug` yet, because those will be stored in a separate collection shown below.
```

   Content module > Languages collection and add two languages.

   ![The table layout for the Languages collection is shown. There are two items displayed. One for English and one for French.](https://marketing.directus.app/assets/9afe044e-405c-46e7-851f-2a400f3672ea.png?key=doc)

6. In the Articles collection, create a new article and scroll down to the Translations field and start translating your
   content.

   You can show one language at a time or show two languages side by side.

   ![A Directus item detail page is shown using the Translations interface. Editable fields for the English and French translations for an article are displayed side-by-side. Fields include Title, Slug, Summary, and Content. ](https://marketing.directus.app/assets/64eee60e-605b-4d2c-809f-83557ed0d5ef.png?key=doc)

   There is also a progress indicator for the status of each language.

   ![A dropdown menu with a progress indicator for English and French languages are shown. The progress is less than half and shown in red.](https://marketing.directus.app/assets/78a895e9-744e-4d33-9d87-3c5ca4c784e2.png?key=doc)

> 

Reviewer: What's missing here is a quick demo of how to interact with this data via the API
